### PR TITLE
Support fully-qualified --package arguments

### LINKED ISSUE #3563: Support fully-qualified `--package` arguments
We [use Kani in zerocopy](https://github.com/google/zerocopy/blob/7c8358e3d9ec9c90269fb64fa6191546ce09a886/.github/workflows/ci.yml#L291), and pass `--package zerocopy` in CI. However, on our 0.7 branch, we are encountering [this failure](https://github.com/google/zerocopy/actions/runs/11144001222/job/30970296888#step:3:303):

```text
error: There are multiple `zerocopy` packages in your project, and the specification `zerocopy` is ambiguous.
Please re-run this command with one of the following specifications:
  path+file:///home/runner/work/zerocopy/zerocopy#0.7.35
  registry+https://github.com/rust-lang/crates.io-index#zerocopy@0.7.35
error: Failed to execute cargo (exit status: 101). Found 0 compilation errors.
```

This seems to be due to Kani having a transitive dependency on zerocopy 0.7. However, Kani does not support passing either suggested syntax as an argument to the `--package` flag. If Kani supported `--package path+file:///home/runner/work/zerocopy/zerocopy#0.7.35`, it would allow us to fix this failure.

### PR DESCRIPTION
**Pull request by celinval**
We were previously matching the name of the package, which would limit users if the name is ambiguous.

Instead, use `cargo pkgid` to validate the `--package` argument and to retrieve the package id.

Resolves #3563 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 and MIT licenses.

### CODE REVIEW

File: kani-driver/src/args/cargo.rs [hunk 1]
```diff
@@ -33,7 +33,7 @@ pub struct CargoCommonArgs {
     #[arg(long)]
     pub workspace: bool,
 
-    /// Run Kani on the specified packages.
+    /// Run Kani on the specified packages  (see `cargo help pkgid` for accepted format)
```
Comment (zhassan-aws) [inline kani-driver/src/args/cargo.rs:36]:
```suggestion
    /// Run Kani on the specified packages (see `cargo help pkgid` for the accepted format)
```

File: kani-driver/src/call_cargo.rs [hunk 2]
```diff
@@ -336,6 +337,93 @@ impl KaniSession {
...
+                reader.read_line(&mut line)?;
+                trace!(package_id=?line, "package_ids");
+                Ok((PackageId { repr: line.trim().to_string() }, pkg.as_str()))
+            })
+            .collect()
+    }
+
+    /// Extract the packages that should be verified.
+    ///
+    /// The result is build following these rules:
```
Comment (zhassan-aws) [inline kani-driver/src/call_cargo.rs:371]:
```suggestion
    /// The result is built following these rules:
```

File: kani-driver/src/call_cargo.rs [hunk 3]
```diff
@@ -336,6 +337,93 @@ impl KaniSession {
...
+                .collect();
+            if filtered.len() < args.cargo.package.len() {
+                // Some packages specified in `--package` were not found in the workspace.
+                let outer: Vec<_> = metadata
+                    .packages
+                    .iter()
+                    .filter_map(|pkg| pkg_ids.get(&pkg.id).copied())
+                    .collect();
+                bail!(
+                    "The following packages specified do not belong to this workspace: `{}`",
```
Comment (zhassan-aws) [inline kani-driver/src/call_cargo.rs:404]:
```suggestion
                    "The following specified packages were not found in this workspace: `{}`",
```

**Review by zhassan-aws — APPROVED** [review]
Thanks! Some nits.

**Comment by celinval** [comment]
Thanks @zhassan-aws! I love when you suggest changes... it makes my life much easier. :smiley:

File: tests/script-based-pre/ambiguous_crate/expected [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+But passing the correct package works
+
+error: There are multiple `zerocopy` packages in your project, and the specification `zerocopy` is ambiguous.
+Please re-run this command with one of the following specifications:
+  zerocopy@0.0.1
+  zerocopy@0.8.4
+error: Failed to retrieve information for `zerocopy`
+Found expected ambiguous crate error
+
+error: The following packages specified do not belong to this workspace: `zerocopy@0.8.4`
```
Comment (celinval) [inline tests/script-based-pre/ambiguous_crate/expected:14]:
```suggestion
error: The following specified packages were not found in this workspace: `zerocopy@0.8.4`
```

**Comment by joshlf** [comment]
Thanks for fixing this! Any idea when this will be released on crates.io?

**Pull request merged**